### PR TITLE
Add POST_CREATE properties

### PR DESCRIPTION
**Pull request by treysp**
Adds mysql properties from #1091 in new POST_CREATE location

### CODE REVIEW

**Comment by treysp** [comment]
This parses and generates the options correctly, but transpile is not respecting case sensitivity in a couple of places.

This test I was going to add to identity doesn't pass:
`CREATE ALGORITHM=UNDEFINED DEFINER=dbt@% SQL SECURITY DEFINER VIEW a AS (SELECT a FROM b)`

It transpiles as:
`CREATE ALGORITHM=UNDEFINED DEFINER=DBT@% SQL SECURITY DEFINER VIEW A AS (SELECT A FROM B)`

I'll come back and work on this later today, but figured I'd get it in a pr in case you want to finish it off before I get back to it.

File: sqlglot/expressions.py [hunk 1]
```diff
@@ -1285,6 +1285,18 @@ class Property(Expression):
     arg_types = {"this": True, "value": True}
 
 
+class AlgorithmProperty(Expression):
+    arg_types = {"this": True}
+
+
+class DefinerProperty(Expression):
+    arg_types = {"this": True}
```
Comment (GeorgeSittas) [inline sqlglot/expressions.py:1293]: You can just do `pass` here.

File: sqlglot/expressions.py [hunk 2]
```diff
@@ -1285,6 +1285,18 @@ class Property(Expression):
...
+
+
+class SqlSecurityProperty(Expression):
+    arg_types = {"definer": True}
```
Comment (GeorgeSittas) [inline sqlglot/expressions.py:1297]: Maybe we can just use `this` for this and then do `pass` too.

**Review by GeorgeSittas — COMMENTED** [review]
Looks good, let's add some tests.

**Comment by GeorgeSittas** [comment]
Seems like I missed your comment about tests etc. Thanks for the PR, I'll try to address remaining comments and merge it.

**Comment by treysp** [comment]
Gah, I totally missed that it was being parsed as a Command due to the _parse_create routing. Sorry for the messy/incomplete PR...

**Comment by GeorgeSittas** [comment]
No worries

File: sqlglot/expressions.py [hunk 3]
```diff
@@ -1428,10 +1442,12 @@ class Properties(Expression):
     arg_types = {"expressions": True}
 
     NAME_TO_PROPERTY = {
```
Comment (tobymao) [inline sqlglot/expressions.py:1444]: i'm not a big fan of the NAME_TO_PROPERTY dictionary. why isn't this just a property on on the property?

File: sqlglot/generator.py [hunk 4]
```diff
@@ -552,6 +556,15 @@ def create_sql(self, expression: exp.Create) -> str:
                 )
             index_sql = "".join(indexes_sql)
 
+        postcreate_props_sql = ""
+        if properties and properties_locs[exp.Properties.Location.POST_CREATE]:
```
Comment (tobymao) [inline sqlglot/generator.py:560]: can you refactor all the places where we do if properties and properties[xxx] and make properties default to a dict

properties = expression.args.get("Propertyers") or {}

then you can just use get

File: sqlglot/parser.py [hunk 5]
```diff
@@ -1129,6 +1138,26 @@ def _parse_with_property(
...
+        if not user or not host:
+            return None
+
+        return exp.DefinerProperty(this=f"{user}@{host}")
+
+    def _parse_sqlsecurity(self) -> exp.Expression:
+        if self._match_text_seq("DEFINER"):
+            definer = True
+        elif self._match_text_seq("INVOKER"):
+            definer = False
```
Comment (tobymao) [inline sqlglot/parser.py:1158]: isn't defined undefined here? if it doesn't match these two branches?

File: sqlglot/expressions.py (hunk 3 above)
Comment (GeorgeSittas) [reply sqlglot/expressions.py:1444]: Agreed, it doesn't feel very elegant. On the other hand, we used to store the property name in `this`, which wasn't very elegant either.

File: sqlglot/parser.py (hunk 5 above)
Comment (GeorgeSittas) [reply sqlglot/parser.py]: fixed

**Comment by GeorgeSittas** [comment]
I'm gonna merge this so that I can iterate faster and address the remaining two comments from Toby.

**Pull request merged**